drivers/hmis/app/models/hmis/role.rb

```diff
@@ -8,8 +8,37 @@ class Hmis::Role < ::ApplicationRecord
   self.table_name = :hmis_roles
   # Warehouse roles do not have a paper trail, so neither do these
 
-  has_many :user_hmis_data_source_roles, class_name: '::Hmis::UserHmisDataSourceRole'
-  has_many :users, through: :user_hmis_data_source_roles, source: :user
+  has_many :access_controls, class_name: '::Hmis::AccessControl', inverse_of: :role
+  has_many :user_access_controls, through: :access_controls
+  has_many :users, through: :user_access_controls
+
+  # has_many :user_hmis_data_source_roles, class_name: '::Hmis::UserHmisDataSourceRole'
+  # has_many :users, through: :user_hmis_data_source_roles, source: :user
+
+  scope :with_all_permissions, ->(*perms) do
+    where(**perms.map { |p| [p, true] }.to_h)
+  end
+
+  scope :with_any_permissions, ->(*perms) do
+    scope = self
+    perms.map { |p| scope = scope.or(Hmis::Role.where(p => true)) }
+    scope
+  end
+
+  scope :with_editable_permissions, -> do
+    with_any_permissions(
+      :can_delete_assigned_project_data,
+      :can_administer_hmis,
+      :can_delete_enrollments,
+    )
+  end
+
+  scope :with_viewable_permissions, -> do
+    with_any_permissions(
+      :can_view_full_ssn,
+      :can_view_clients,
+    )
+  end
```

Comment:
I think it'll be better long-term to decorate the `permissions_with_descriptions` below with if the permission is editable or viewable, something like:
```
can_view_full_ssn: {
        description: 'Allow the user to see client\'s full SSN.',
        administrative: false,
        access: [:viewable],
        categories: [
          'Client Details',
        ],
      },
```
and then make a method that returns an array of the keys for editable or viewable something like:
```
def permissions_for_access(access_level)
  permissions_with_descriptions.select { |m| m[: access].include?(access_level) }.keys
end
```
That will centralize the identification of access and allow a permission to be both a view and edit (or other type of access in the future.)